src/fourier.jl

```diff
@@ -0,0 +1,773 @@
+            unnorm_power = keepat!(unnorm_power,fgt0)
+            if return_auxil
+                unnorm_pd1 = keepat!(unnorm_pd1,fgt0)
+                unnorm_pd2 = keepat!(unnorm_pd2,fgt0)
+            end
+        end
+
+        cs_seg = unnorm_power
+        p1_seg = unnorm_pd1
+        p2_seg = unnorm_pd2
+
+        # If normalization has to be done interval by interval, do it here.
+        if !(use_common_mean)
+            mean1 = n_ph1 / n_bin
+            mean2 = n_ph2 / n_bin
+            mean = n_ph / n_bin
+            variance = nothing
+
+            if !isnothing(variance1)
+                variance = sqrt(variance1 * variance2)
+            end
+
+            cs_seg = normalize_periodograms(
+                unnorm_power, dt, n_bin; mean_flux = mean, n_ph=n_ph, norm=norm,
+                power_type=power_type, variance=variance
+            )
+
+            if return_auxil
+                p1_seg = normalize_periodograms(
+                    unnorm_pd1, dt, n_bin; mean_flux = mean1, n_ph=n_ph1, norm=norm,
+                    power_type=power_type, variance=variance1
+                )
+                p2_seg = normalize_periodograms(
+                    unnorm_pd2, dt, n_bin; mean_flux = mean2, n_ph=n_ph2, norm=norm,
+                    power_type=power_type, variance=variance2
+            )
+            end
+        end
+        # Initialize or accumulate final averaged spectra
+        cross = sum_if_not_none_or_initialize(cross, cs_seg)
+        unnorm_cross = sum_if_not_none_or_initialize(unnorm_cross,
+                                                     unnorm_power)
+
+        if return_auxil
+            unnorm_pds1 = sum_if_not_none_or_initialize(unnorm_pds1,
+                                                        unnorm_pd1)
+            unnorm_pds2 = sum_if_not_none_or_initialize(unnorm_pds2,
+                                                        unnorm_pd2)
+            pds1 = sum_if_not_none_or_initialize(pds1, p1_seg)
+            pds2 = sum_if_not_none_or_initialize(pds2, p2_seg)
+        end
+
+        n_ave += 1
+    end
+
+    # If no valid intervals were found, return only `None`s
+    if isnothing(cross)
+        return nothing
+    end
+
+    # Calculate the mean number of photons per chunk
+    n_ph1 = sum_of_photons1 / n_ave
+    n_ph2 = sum_of_photons2 / n_ave
+    n_ph = sqrt(n_ph1 * n_ph2)
+
+    # Calculate the common mean number of photons per bin
+    common_mean1 = n_ph1 / n_bin
+    common_mean2 = n_ph2 / n_bin
+    common_mean = n_ph / n_bin
+
+    if !isnothing(common_variance1)
+        # Note: the variances we summed were means, not sums. Hence M, not M*N
+        common_variance1 /= n_ave
+        common_variance2 /= n_ave
+        common_variance = sqrt(common_variance1 * common_variance2)
+    end
+
+    # Transform the sums into averages
+    cross /= n_ave
+    unnorm_cross /= n_ave
+    if return_auxil
+        unnorm_pds1 /= n_ave
+        unnorm_pds2 /= n_ave
```

Comment:
With
```suggestion
        unnorm_pds1 ./= n_ave
        unnorm_pds2 ./= n_ave
```
you modify `unnorm_pds1` and `unnorm_pds2` in-place without allocating temporary arrays and assigning them to these variables.

Reply:
So any place I work with operation on both scalar and vectors, I should use dot broadcast right? (Shouldn't be this then the natural behaviour)